hamilton/function_modifiers/base.py

```diff
@@ -228,6 +228,63 @@ def transform_dag(
         pass
 
 
+class NodeInjector(SubDAGModifier, abc.ABC):
+    """Injects a value as a source node in the DAG. This is a special case of the SubDAGModifier,
+    which gets all the upstream (required) nodes from the subdag and gives the decorator a chance
+    to inject values into them."""
```

Comment:
I'm not sure I'm following. What is the subdag consisting of in this case? Why are we injecting values into required nodes?

Reply:
If we want to load_from some source, that needs to attach to a dependency in the DAG. Thus a "required" node. 

Note that this would be `user_defined` if we didn't inject. 

For instance:

```python
def processed_data(data: pd.DataFrame) -> pd.DataFrame:
    ...
```

Creates a DAG with one node (`processed_data`) and one external input (`data`) -- this we can attach to a `NodeInjector`. I created this as an abstraction as it felt more readable and fit along with the way we're doing other things. I also have a hunch that we'll really want `NodeInjector`s for the LLM case.

Reply:
Yeah some mental context in the docs like this would be great.